src/daemon/Display.cpp

```diff
@@ -386,12 +387,14 @@ namespace SDDM {
         env.insert(QStringLiteral("XDG_SEAT_PATH"), daemonApp->displayManager()->seatPath(seat()->name()));
         env.insert(QStringLiteral("XDG_SESSION_PATH"), daemonApp->displayManager()->sessionPath(QStringLiteral("Session%1").arg(daemonApp->newSessionId())));
         env.insert(QStringLiteral("DESKTOP_SESSION"), session.desktopSession());
-        env.insert(QStringLiteral("XDG_CURRENT_DESKTOP"), session.desktopNames());
+        if (!session.desktopNames().isEmpty())
+            env.insert(QStringLiteral("XDG_CURRENT_DESKTOP"), session.desktopNames());
         env.insert(QStringLiteral("XDG_SESSION_CLASS"), QStringLiteral("user"));
         env.insert(QStringLiteral("XDG_SESSION_TYPE"), session.xdgSessionType());
         env.insert(QStringLiteral("XDG_SEAT"), seat()->name());
         env.insert(QStringLiteral("XDG_VTNR"), QString::number(m_lastSession.vt()));
-        env.insert(QStringLiteral("XDG_SESSION_DESKTOP"), session.desktopNames());
+        if (!session.desktopNames().isEmpty())
+            env.insert(QStringLiteral("XDG_SESSION_DESKTOP"), session.desktopNames());
```

Comment:
This doesn't seem right though (and wasn't right before the change either).

The [specification for the variable](https://www.freedesktop.org/software/systemd/man/pam_systemd.html#desktop=) is a bit vague and systemd-specific, so it's one of the cases where the behavior of other DMs should be taken into account.
Both [LightDM](https://github.com/canonical/lightdm/blob/7e3b2b79d26a21958b7217968dbe74ce2e2bc290/src/seat.c#L979) and [GDM](https://gitlab.gnome.org/GNOME/gdm/-/blob/ae75f642859dcfdcfcaaffc617ebe7a68cf0bed7/daemon/gdm-session.c#L2687) handle `XDG_SESSION_DESKTOP` exactly the same way as `DESKTOP_SESSION`. I.e. it's always set to a name of session file sans '.desktop' extension.

If there's an issue with xinitrc and `XDG_SESSION_DESKTOP` being set, it should be solved somewhere else in an uniform way for all display managers.

The above does not apply to `XDG_CURRENT_DESKTOP`; the change for that variable is correct.

Reply:
Ok, maybe guard this `env.insert` into something like `#if HAVE_SYSTEMD`?